cmd/compute-domain-kubelet-plugin/device_state.go

```diff
@@ -491,6 +492,29 @@ func (s *DeviceState) applyComputeDomainDaemonConfig(ctx context.Context, config
 		ComputeDomain: config.DomainID,
 	}
 
+	cd, err := s.computeDomainManager.GetComputeDomain(ctx, config.DomainID)
+	if err != nil {
+		return nil, fmt.Errorf("error getting compute domain: %w", err)
+	}
+	if cd == nil {
+		return nil, fmt.Errorf("compute domain not found: %s", config.DomainID)
+	}
+
+	// Always inject Compute Domain details into the CD daemon, in a
+	// heterogeneous CD clique ID below may be empty (while other CD details are
+	// set, and consumed by the CD daemon).
+	edits := &cdiapi.ContainerEdits{
+		ContainerEdits: &cdispec.ContainerEdits{
+			Env: []string{
+				fmt.Sprintf("CLIQUE_ID=%s", s.computeDomainManager.cliqueID),
+				fmt.Sprintf("COMPUTE_DOMAIN_UUID=%s", cd.UID),
+				fmt.Sprintf("COMPUTE_DOMAIN_NAME=%s", cd.Name),
+				fmt.Sprintf("COMPUTE_DOMAIN_NAMESPACE=%s", cd.Namespace),
+			},
+		},
+	}
```

Comment:
Let's encapuslate this in a separate function on the `computeDomainManager` and rename computeDomainDaemonSettings.GetCDIContainerEdits to reflect that its specific to when IMEX is involved.